### PR TITLE
Procedure for updating Operator

### PR DESCRIPTION
**Pull request by ravindk89**
# Summary

This is a core upgrade procedure for the MinIO Operator. It focuses on using ``kubectl minio init`` as a baseline.

It also imports Oscar's work for the 3.X.X -> 4.2.2 and 4.2.2 -> 4.2.3 paths, for older Operator deployments.

I've tried to test what I can, and I believe this covers the main considerations.

There is a niche case for Tenants where the user:

1) deployed via YAML only, without specifying an image
2) Continue to upgrade MinIO Operator without every modifying that Tenant YAML again
3) Have their Tenants upgraded along with the Operator

However, this should only be an issue for very old Tenants deployed specifically via CLI. At most we might add something to the 3.X.X upgrade path to ask users to make those fixes themselves, but the included step to upgrade to latest stable should resolve this regardless.

Staged view is available in the #minio-docs channel

### CODE REVIEW

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst [hunk 1]
```diff
@@ -11,6 +11,299 @@ Upgrade MinIO Operator
...
+               accessModes:
+                  - ReadWriteOnce
+               resources:
+                  requests:
+                  storage: 1Ti
+            securityContext:
+            runAsUser: 0
+            runAsGroup: 0
+            runAsNonRoot: false
+            fsGroup: 0
```
Comment (ravindk89) [inline source/operations/install-deploy-manage/upgrade-minio-operator.rst:254]:
```suggestion
            securityContext:
              runAsUser: 0
              runAsGroup: 0
              runAsNonRoot: false
              fsGroup: 0
```

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst [hunk 2]
```diff
@@ -11,6 +11,299 @@ Upgrade MinIO Operator
...
```
Comment (ravindk89) [inline source/operations/install-deploy-manage/upgrade-minio-operator.rst:271]:
```suggestion
              runAsUser: 0
              runAsGroup: 0
              runAsNonRoot: false
              fsGroup: 0
```

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/operations/install-deploy-manage/upgrade-minio-operator.rst:254]: Looks like the YAML in general got messed up. Will fix.

File: source/reference/kubectl-minio-plugin/kubectl-minio-init.rst [hunk 3]
```diff
@@ -18,7 +18,8 @@ Description
 
 .. start-kubectl-minio-init-desc
 
-The :mc:`kubectl minio init` command initializes the MinIO Operator. 
+The :mc:`kubectl minio init` command initializes the MinIO Operator.
+You can also use this command to upgrade an existing Operator installation to the version of the plugin.
```
Comment (djwfyi) [inline source/reference/kubectl-minio-plugin/kubectl-minio-init.rst:22]: This sentence reads awkward to me.
Maybe strike the last part `to the version of the plugin.`?

> You can also use this command to upgrade an Operator installation to a later version that matches the plugin.

I dunno. But that's a suggestion.

**Review by djwfyi — CHANGES_REQUESTED** [review]
Can we get the dummy text in the screenshot to have a version later than last February?

I also suggest doing something to break out the prerequisites from the procedures a bit more. They run together visually right now.

And some other fixes and suggestions throughout. I'll take another pass at it.

**Review by Alevsk — APPROVED** [review]

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst [hunk 4]
```diff
@@ -11,6 +10,321 @@ Upgrade MinIO Operator
...
+   #. Validate the Tenant ``tenant.spec.zones`` values
+
+      Use the following command to validate the specification for each managed MinIO Tenant:
+
+      .. code-block:: shell
+         :class: copyable
+
+         kubectl get tenants <TENANT-NAME> -n <TENANT-NAMESPACE> -o yaml
+
+      - Ensure each ``tenant.spec.zones`` element has a ``names`` field set to the name for that zone.
```
Comment (Alevsk) [inline source/operations/install-deploy-manage/upgrade-minio-operator.rst:247]:
```suggestion
      - Ensure each ``tenant.spec.zones`` element has a ``name`` field set to the name for that zone.
```

Comment (Alevsk) [reply source/operations/install-deploy-manage/upgrade-minio-operator.rst:247]: this one should be singular i think

**Review by Alevsk — CHANGES_REQUESTED** [review]

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst (hunk 4 above)
Comment (djwfyi) [inline source/operations/install-deploy-manage/upgrade-minio-operator.rst:247]: These two bullet points are also being bolded by the new css you added.

**Review by djwfyi — APPROVED** [review]
One fix, one heading suggestion.
And some unintended CSS inheritance from the style you added.

Otherwise, LGTM.

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst [hunk 5]
```diff
@@ -11,6 +11,299 @@ Upgrade MinIO Operator
...
+Upgrade MinIO Operator 4.2.2 to |operator-version-stable|
+---------------------------------------------------------
+
+This procedure assumes that:
+
+- You have an existing MinIO Operator deployment running 4.2.2
+- Your Kubernetes cluster runs 1.19.0 or later
+- Your local host has ``kubectl`` installed and configured with access to the Kubernetes cluster
+
+1. (Optional) Update each MinIO Tenant to the latest stable MinIO Version.
```
[the remark this answers is not in the document]
Comment (djwfyi) [reply source/operations/install-deploy-manage/upgrade-minio-operator.rst]: There are some unintended consequences of the new CSS.

[the remark this answers is not in the document]
Comment (djwfyi) [reply source/operations/install-deploy-manage/upgrade-minio-operator.rst]: But the italicizing worked out well even with the bolding.

File: source/operations/install-deploy-manage/upgrade-minio-operator.rst [hunk 6]
```diff
@@ -11,6 +11,299 @@ Upgrade MinIO Operator
...
+      kubectl get pods -l "v1.min.io/tenant" --all-namespaces
+
+#. Upgrade to |operator-version-stable|
+
+   Follow the :ref:`minio-k8s-upgrade-minio-operator-procedure` procedure to upgrade to the latest stable Operator version.
+
+Upgrade MinIO Operator 3.X.X to |operator-version-stable|
+---------------------------------------------------------
+
+This procedure upgrades MinIO Operator 3.X.X deployments to Operator 4.2.2.
```
[the remark this answers is not in the document]
Comment (djwfyi) [reply source/operations/install-deploy-manage/upgrade-minio-operator.rst]: New approach is good 👍

File: source/_static/scss/includes/_reset.scss [hunk 7]
```diff
@@ -316,4 +315,16 @@ abbr, acronym {
...
+}
+
+// ----------------------
+// Container - Procedure
+// ----------------------
+
+.container.procedure {
+   padding: 0px;
+
+   ol li p:first-of-type {
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/_static/scss/includes/_reset.scss:327]: Fixed!

**Pull request merged**